extensions/ipynb/src/notebookImagePaste.ts

```diff
@@ -66,19 +66,68 @@ class CopyPasteEditProvider implements vscode.DocumentPasteEditProvider {
 
 		return { insertText: pasteSnippet, additionalEdit: workspaceEdit };
 	}
+}
 
-	private getCellFromCellDocument(cellDocument: vscode.TextDocument): vscode.NotebookCell | undefined {
-		for (const notebook of vscode.workspace.notebookDocuments) {
-			if (notebook.uri.path === cellDocument.uri.path) {
-				for (const cell of notebook.getCells()) {
-					if (cell.document === cellDocument) {
-						return cell;
-					}
-				}
-			}
+class ImageDropEditProvider implements vscode.DocumentDropEditProvider {
+	async provideDocumentDropEdits(
+		document: vscode.TextDocument,
+		_position: vscode.Position,
+		dataTransfer: vscode.DataTransfer,
+		_token: vscode.CancellationToken
+	): Promise<vscode.DocumentDropEdit | undefined> {
+		const enabled = vscode.workspace.getConfiguration('ipynb', document).get('experimental.pasteImages.enabled', false);
```

Comment:
For the future, this likely should have its own setting (or the existing setting should have a more generic name)